src/gmail_handler.py

```diff
@@ -165,11 +198,162 @@ def get_sender(self, message, interest):
+
+        try:
+
+            # Get the full message
+
+            message = (
+                self.google_client.service.users()
+                .messages()
+                .get(userId=user_id, id=msg_id, format='full')
+                .execute()
+            )
+
+            # Create or check the download directory
+
+            if not os.path.exists(download_dir):
+
+                os.makedirs(download_dir)
+
+            # Extract message headers
+
+            headers = message.get('payload', {}).get('headers', [])
+
+            subject = next(
+                (
+                    header['value']
+                    for header in headers
+                    if header['name'].lower() == 'subject'
+                ),
+                'No Subject',
+            )
+
+            date = next(
+                (
+                    header['value']
+                    for header in headers
+                    if header['name'].lower() == 'date'
+                ),
+                'Unknown Date',
+            )
+
+            cleaned_subject = ''.join([c if c.isalnum() else '_' for c in subject])
+
+            filename_prefix = f'{download_dir}/{cleaned_subject}_{msg_id}'
+
+            # Save metadata (subject, date)
+
+            with open(f'{filename_prefix}.meta.txt', 'w') as meta_file:
+
+                meta_file.write(f'Subject: {subject}\n')
+
+                meta_file.write(f'Date: {date}\n\n')
+
+                meta_file.write(
+                    f'Snippet: {message.get('snippet', 'No snippet available.')}\n'
+                )
+
+            # Download attachments and save
+
+            self._process_message_parts(
+                message.get('payload', {}), filename_prefix, msg_id
+            )
+
+            return f'Message {msg_id} downloaded successfully to {filename_prefix}.'
+
+        except errors.HttpError as error:
+
+            return f'Error downloading message {msg_id}: {error}'
+
+    def _process_message_parts(self, payload, filename_prefix, msg_id):
+        """Processes parts of a message recursively to download attachments."""
+
+        if 'parts' in payload:
+
+            for part in payload['parts']:
+
+                self._process_message_parts(part, filename_prefix)
+
+        # Download attachments
+
+        if payload.get('filename'):
+
+            attachment_id = payload.get('body', {}).get('attachmentId')
+
+            if attachment_id:
+
+                attachment = (
+                    self.google_client.service.users()
+                    .messages()
+                    .attachments()
+                    .get(userId='me', messageId=msg_id, id=attachment_id)
+                    .execute()
+                )
+
+                data = base64.urlsafe_b64decode(attachment['data'])
+
+                with open(
+                    f'{filename_prefix}_{payload['filename']}', 'wb'
+                ) as attachment_file:
+
+                    attachment_file.write(data)
```

Comment:
```suggestion
        if payload.get('filename'):
            attachment_id = payload.get('body', {}).get('attachmentId')
            if attachment_id:
                attachment = (
                    self.google_client.service.users()
                    .messages()
                    .attachments()
                    .get(userId='me', messageId=msg_id, id=attachment_id)
                    .execute()
                )
                data = base64.urlsafe_b64decode(attachment['data'])

                with open(
                    f'{filename_prefix}_{payload['filename']}', 'wb'
                ) as attachment_file:
                    attachment_file.write(data)
```

Really no need for so much spaces. If you could wrap it up a bit it would be great. Also in other places.